core/dictgen/src/rootcling_impl.cxx

```diff
@@ -772,9 +772,19 @@ string gLibsNeeded;
 
 ////////////////////////////////////////////////////////////////////////////////
 
-void RecordDeclCallback(const char *c)
+void RecordDeclCallback(const clang::RecordDecl* recordDecl)
 {
-   string need(gAutoloads[c]);
+   std::string need;
+   if (recordDecl->hasOwningModule()) {
+      clang::Module *M = recordDecl->getOwningModule()->getTopLevelModule();
+      need = "lib" + M->Name + gLibraryExtension;
+   } else {
+      std::string qual_name;
+      RScanner::GetDeclQualName(recordDecl, qual_name);
+
+      need = gAutoloads[qual_name.c_str()];
```

Comment:
The temporary std::string probably introduces lifetime issues. Maybe we should make gAutoloads store string copies.

Reply:
Can this go into different commit/PR?

Reply:
It seems to be introduced by this PR, so it should be here.